File: LeetCode/Study Plan/Practice Problems/Medium/Gas Station/Go/gas_station.go
package main

import "fmt"

func gas_station(gas []int, cost []int) int {
	curr_tank, total_tank, start := 0, 0, 0
	for i := 0; i < len(gas); i++ {
		total_tank += gas[i] - cost[i]
		curr_tank += gas[i] - cost[i]
		if curr_tank < 0 {
			start = i + 1
			curr_tank = 0
		}
	}
	if total_tank < 0 {
		return -1
	}
	return start
}

func main() {
	fmt.Println(gas_station([]int{1, 2, 3, 4, 5}, []int{3, 4, 5, 1, 2}))
}

File: LeetCode/Study Plan/Practice Problems/Medium/Gas Station/Java/gasStation.java
class Solution {
    public int canCompleteCircuit(int[] gas, int[] cost) {
        int size = gas.length;
        int currTank = 0, totalTank = 0;
        int start = 0;
        for (int i = 0; i < size; i++) {
            totalTank += gas[i]-cost[i];
            currTank += gas[i]-cost[i];
            if (currTank < 0) {
                start = i+1;
                currTank = 0;
            }
        }
        if (totalTank < 0) return -1;
        return start;
    }
}
